src/services/shop/shop-router.ts

```diff
@@ -42,148 +43,423 @@ shopRouter.get(
-        path: "/shop/item/{id}/",
+        method: "post",
+        path: "/shop/cart/{itemId}/",
         tag: Tag.SHOP,
-        role: Role.ADMIN,
-        summary: "Updates a shop item",
+        role: Role.USER,
+        summary: "Adds item to users cart",
         parameters: z.object({
-            id: ShopItemIdSchema,
+            itemId: ShopItemIdSchema,
         }),
-        body: ShopItemUpdateRequestSchema,
         responses: {
             [StatusCode.SuccessOK]: {
-                description: "The new item",
-                schema: ShopItemSchema,
+                description: "The successfully updated order",
+                schema: ShopOrderInfoSchema,
             },
             [StatusCode.ClientErrorNotFound]: {
-                description: "Item doesn't exist",
+                description: "Shop Item DNE",
                 schema: ShopItemNotFoundErrorSchema,
             },
+            [StatusCode.ClientErrorBadRequest]: {
+                description: "Not enough quantity in shop",
+                schema: ShopInsufficientQuantityErrorSchema,
+            },
+            [StatusCode.ClientErrorPaymentRequired]: {
+                description: "Not enough points to purchase",
+                schema: ShopInsufficientFundsErrorSchema,
+            },
+            [StatusCode.ServerErrorInternal]: {
+                description: "Errors that should never happen",
+                schema: ShopInternalErrorSchema,
+            },
         },
     }),
     async (req, res) => {
-        const { id: itemId } = req.params;
-        const updateRequest = req.body;
+        const { itemId } = req.params;
+        const { id: userId } = getAuthenticatedUser(req);
+
+        let userOrder = await Models.ShopOrder.findOne({ userId: userId });
+        //user doesn't have a order yet
+        if (!userOrder) {
+            const shopOrder: ShopOrder = {
+                items: [],
+                quantity: [],
+                userId: userId,
+            };
+
+            userOrder = await Models.ShopOrder.create(shopOrder);
+        }
 
-        const updatedItem = await Models.ShopItem.findOneAndUpdate({ itemId }, updateRequest, {
-            new: true,
-        });
+        // This should never get hit, just checking here so typescript doesn't get mad
+        if (!userOrder) {
+            return res.status(StatusCode.ServerErrorInternal).send(ShopInternalError);
+        }
 
-        if (!updatedItem) {
+        //check if enough quantity in shop
+        const item = await Models.ShopItem.findOne({ itemId: itemId });
+        if (!item) {
             return res.status(StatusCode.ClientErrorNotFound).send(ShopItemNotFoundError);
         }
 
-        const withoutInstances = {
-            ...updatedItem.toObject(),
-            instances: undefined,
-        };
+        if (item.quantity <= 0) {
+            return res.status(StatusCode.ClientErrorBadRequest).send(ShopInsufficientQuantityError);
+        }
```

Comment:
This check isn't enough - we have to check all items in the order whenever the order is updated. The same goes for verifying points. Otherwise, this will let them add 1000 of the same item since it only checks the singular item. They won't be able to buy it, but that's not a good experience.